### PR TITLE
[CDAP-15787] Adding validation to Datastore source/sink, BigTable source/sink, GCS action plugins

### PR DESCRIPTION
**Pull request by CuriousVini**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java [hunk 1]
```diff
@@ -104,12 +104,12 @@ public boolean isAllowSchemaRelaxation() {
   public void validate(FailureCollector collector) {
     super.validate(collector);
     String bucket = getBucket();
-    if (!containsMacro("bucket") && bucket != null) {
+    if (!containsMacro(NAME_BUCKET) && bucket != null) {
       // Basic validation for allowed characters as per https://cloud.google.com/storage/docs/naming
       Pattern p = Pattern.compile("[a-z0-9._-]+");
       if (!p.matcher(bucket).matches()) {
         collector.addFailure("Bucket must only contain lowercase characters, numbers,'.', '_', and '-'", null)
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java]: existing, but missing a space before '.'

File: src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java [hunk 2]
```diff
@@ -91,40 +88,49 @@ public BigtableSink(BigtableSinkConfig config) {
...
-        validateExistingTable(connection, tableName);
+        validateExistingTable(connection, tableName, collector);
       } else {
-        createTable(connection, tableName);
+        createTable(connection, tableName, collector);
       }
     } catch (IOException e) {
-      throw new InvalidStageException("Failed to connect to Bigtable", e);
+      collector.addFailure(
+        String.format("Failed to prepare configuration for job : %s", e.getMessage()), null)
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java]: how come the error message was changed? Is it not always the case the it's some issue connecting to Bigtable?

File: src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java [hunk 3]
```diff
@@ -152,36 +160,41 @@ private Configuration getConfiguration() {
...
         String.format("Field '%s' declared as key alias does not exist in input schema", config.keyAlias),
-        BigtableSinkConfig.KEY_ALIAS
-      );
+        "Specify input field name as key alias.").withConfigProperty(BigtableSinkConfig.KEY_ALIAS);
     }
     List<Schema.Field> fields = inputSchema.getFields();
     if (fields == null || fields.isEmpty()) {
-      throw new InvalidStageException("Input schema should contain fields");
+      collector.addFailure("Input schema does not contain any fields.",
+                           "Input schema must contain fields map.");
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java]: In general, I don't think we need a corrective action when the message is saying that something required doesn't exist.

File: src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java (hunk 2 above)
Comment (CuriousVini) [reply src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java]: yea that should not be changed, fixed.

**Review by albertshau — APPROVED** [review]

**Pull request merged**